### ISSUE TITLE
Better type documentation for components

### ISSUE THREAD

**Issue by mhkeller** [mhkeller/layercake#58]
Currently, all of the components have jsdoc comments but the vs code extension doesn't seem to recognize them well.

The component description only gets pulled in (when hovering over the component's import statement) if the format is such where the `@type` definitions are included in one big comment at the top immediately above the exported variables like this:

```js
<script>
  /**
    Generates an SVG marker containing a marker for a triangle makes a nice arrowhead. Add it to the named slot called "defs" on the SVG layout component.
    @type {String} [fill='#000'] – The arrowhead's fill color.
    @type {String} [stroke='#000'] – The arrowhead's fill color.
  */
  export let fill = '#000';
  export let stroke = '#f0c';
</script>

<marker id="arrowhead" viewBox="-10 -10 20 20" markerWidth="17" markerHeight="17" orient="auto">
  <path d="M-6,-6 L 0,0 L -6,6" fill="{fill}" stroke="{stroke}"/>
</marker>
```

This yields a popup when imported like this, which is helpful because it has the comment description. It has `undefined` for the props though:

<img width="905" alt="Screen Shot 2021-12-21 at 12 16 42 AM" src="https://user-images.githubusercontent.com/498744/146875326-51633a36-3325-4fbf-a8bc-68c45c09afc5.png">

In that format the popups over each individual property are also undefined and it pulls in that entire first comment, which isn't great:

<img width="876" alt="Screen Shot 2021-12-21 at 12 17 17 AM" src="https://user-images.githubusercontent.com/498744/146875385-d697bb71-438e-45e3-8354-65e06749ed73.png">

When the component is formatted like this:

```js
<script>
  /**
    Generates an SVG marker containing a marker for a triangle makes a nice arrowhead. Add it to the named slot called "defs" on the SVG layout component.
  */
  /** @type {String} [fill='#000'] – The arrowhead's fill color. */
    export let fill = '#000';
  /** @type {String} [stroke='#000'] – The arrowhead's fill color. */
  export let stroke = '#f0c';
</script>

<marker id="arrowhead" viewBox="-10 -10 20 20" markerWidth="17" markerHeight="17" orient="auto">
  <path d="M-6,-6 L 0,0 L -6,6" fill="{fill}" stroke="{stroke}"/>
</marker>
```

The top-level hover now loses the main comment, but the individual props are nicely annotated:
<img width="867" alt="Screen Shot 2021-12-21 at 12 18 25 AM" src="https://user-images.githubusercontent.com/498744/146875526-7b97342e-fa93-4e8c-b1a5-c8eec7c72b0a.png">
<img width="709" alt="Screen Shot 2021-12-21 at 12 18 33 AM" src="https://user-images.githubusercontent.com/498744/146875529-0bf03225-858c-46ad-9f32-474d38b54147.png">

But it still shows a lot of `undefined` fields.

Any thoughts?

**Comment by dummdidumm** [comment]
Do you have another extension installed which shows different hovers by any chance? For me, using VS Code with the official Svelte extension, it works as expected:
![image](https://user-images.githubusercontent.com/5968653/147262016-dd6b8625-2272-49d6-9bed-3f5a3c3d2aa6.png)

Documenting the default import of a Svelte file (the component): https://svelte.dev/faq#how-do-i-document-my-components

**Comment by mhkeller** [comment]
Thanks @dummdidumm. That helped me figure out it was a Svelte intellisense plugin I had. I've documented the components now like this, following that link: https://github.com/mhkeller/layercake/blob/835941a2e303904411c3e4758d5f28e254de964c/src/components/AxisX.svelte#L9

It doesn't say whether to document the props also in that html comment block. It works better if you add the comments inline for each prop. They don't show up when you hover over the component itself, but you get better hovers when you hover over each prop...

**Comment by dummdidumm** [comment]
Documenting the props like shown in your code snippet is correct. To also show them when hovering on the component itself - that's not supported yet, there's an issue for this where you can voice your opinion: https://github.com/sveltejs/language-tools/issues/1308

**Comment by mhkeller** [comment]
Awesome, great to know!

**Comment by mhkeller** [comment]
This is now merged and released as 6.0.0 https://github.com/mhkeller/layercake/releases/tag/v6.0.0
